src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,28 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final CopyOnWriteArrayList<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        executorService.shutdown();
```

Comment:
Depending on the implementation, keep in mind that this could also be required:

```suggestion
        listeners.clear();
        executorService.shutdown();
```